robot_modeling/DQ_SerialManipulatorDH.m

```diff
@@ -224,12 +321,15 @@
             % This function does not take into account any base or
             % end-effector displacements and should be used mostly
             % internally in DQ_kinematics
+            obj.check_q_vec(q);
             
             if nargin < 3
-                to_ith_link = obj.n_links;
+                to_ith_link = obj.dim_configuration_space_;
```

Comment:
In order to "simplify" the code, a bad practice was done here, which does not necessarily translate well to different languages: if there are fewer than three parameters, it is assumed that the parameter `to_ith_link` will not be used, hence it is safe to instantiate a local variable with the same name. This practice is confusing, hence it is better to use the previous implementation.

Reply:
@bvadorno That was AFAIK introduced by me in 20.04 and slipped through our review. I didn’t know it was considered a bad practice in MATLAB, so I’ll keep that in mind. 

In other languages, it would be handled as follows. Regardless of what I wrote below, we must keep the code consistent so @juanjqo please fix this one and double check if I haven’t used the same logic in other places. 

- It is commonplace in Python to default the argument as None and give it a value inside the method for arguments such as the one in question that cannot be evaluated when the module is loaded. (e.g., some of the examples in Section 2.12.4 https://google.github.io/styleguide/pyguide.html)
- In CPP we do not need to do this because we can overload the method, and that’s how it’s implemented there.

Reply:
@bvadorno @mmmarinho I did the requested modifications. Now, the method raw_pose_jacobian looks like:

```Matlab
            obj.check_q_vec(q);
            if nargin == 3
                n = to_ith_link;
            else
                n = obj.dim_configuration_space_;
            end

            obj.check_to_ith_link(n);
            x_effector = obj.raw_fkm(q,n);
          
            x = DQ(1);
            J = zeros(8,n);
            
            for i = 1:n
                w = obj.get_w(i);
                z = 0.5*Ad(x,w);
                x = x*obj.dh2dq(q(i),i);
                j = z * x_effector;
                J(:,i) = vec8(j);
            end
```